protocols/imap/src/main/scala/org/apache/james/imap/processor/SearchProcessor.scala

```diff
@@ -0,0 +1,176 @@
+
+          } else {
+            // Just save the returned sequence-set as this is not SEARCHRES + ESEARCH
+            SearchResUtil.saveSequenceSet(session, idRanges.toArray)
+            new SearchResponse(ids.toArray, highestModSeq)
+          }
+        }
+      responder.respond(response)
+      val omitExpunged = !useUids
+      unsolicitedResponses(session, responder, omitExpunged, useUids)
+      okComplete(request, responder)
+    } catch {
+      case e: MessageRangeException =>
+        SearchProcessor.LOGGER.debug("Search failed in mailbox {} because of an invalid sequence-set ", session.getSelected.getMailboxId, e)
+        taggedBad(request, responder, HumanReadableText.INVALID_MESSAGESET)
+      case e: MailboxException =>
+        SearchProcessor.LOGGER.error("Search failed in mailbox {}", session.getSelected.getMailboxId, e)
+        no(request, responder, HumanReadableText.SEARCH_FAILED)
+        if (resultOptions.contains(SearchResultOption.SAVE)) { // Reset the saved sequence-set on a BAD response if the SAVE option was used.
+          // See RFC5182 2.1.Normative Description of the SEARCHRES Extension
+          SearchResUtil.resetSavedSequenceSet(session)
+        }
+    } finally session.setAttribute(SearchProcessor.SEARCH_MODSEQ, null)
+  }
+
+  private def asResults(session: ImapSession, useUids: Boolean, uids: Seq[MessageUid]) =
+    if (useUids)
+      uids.map(_.asLong)
+    else
+      uids.map(session.getSelected.msn(_))
+        .filter(_ != SelectedMailbox.NO_SUCH_MESSAGE)
+        .map(_.longValue)
+
+  @throws[MailboxException]
+  private def performUidSearch(mailbox: MessageManager, query: SearchQuery, msession: MailboxSession) = try {
+    import scala.jdk.StreamConverters._
+    val stream = mailbox.search(query, msession)
+    try stream.toScala(Seq)
+    finally if (stream != null) stream.close()
+  }
+
+  /**
+   * Find the highest mod-sequence number in the given {@link MessageRange}'s.
+   */
+  @throws[MailboxException]
+  private def findHighestModSeq(session: MailboxSession, mailbox: MessageManager, ranges: Seq[MessageRange], currentHighest: ModSeq): ModSeq = {
+    var highestModSeq: ModSeq = null
+    // Reverse loop over the ranges as its more likely that we find a match at the end
+    for (range <- ranges.reverse) {
+      val results = mailbox.getMessages(range, FetchGroup.MINIMAL, session)
+      while (results.hasNext) {
+        val modSeq = results.next.getModSeq
+        highestModSeq =
+          if (highestModSeq == null || modSeq.asLong > highestModSeq.asLong)
+            modSeq
+          else
+            highestModSeq
+        if (highestModSeq.equals(currentHighest)) return highestModSeq
+      }
+    }
+    highestModSeq
+  }
+
+  override def getImplementedCapabilities(session: ImapSession): util.List[Capability] = {
+    import scala.jdk.CollectionConverters._
```

Comment:
So far my team did these imports at the class level.

Any reasons to do it at the scope level?

Reply:
Yes, this import actually put some implicit conversion into the scope, it's a good practice to limit the scope of implicits for readability and compile-time.